### PR TITLE
PS-3812 Search widget for concrete products on Shopping List and Cart

### PR DESCRIPTION
**Pull request by Compixx**
Core PR: https://github.com/spryker/spryker/pull/4794

### CODE REVIEW

File: src/Spryker/Zed/EventBehavior/Business/EventBehaviorFacade.php [hunk 1]
```diff
@@ -109,6 +109,26 @@ public function getAvailableResourceNames(): array
...
 
+    /**
+     * {@inheritdoc}
+     *
+     * @api
+     *
+     * @param \Generated\Shared\Transfer\EventEntityTransfer[] $eventTransfers
+     * @param string $foreignKeyColumnName
+     *
+     * @return array ['foreignKey' => ['relatedForeignKeyName1' => relatedForeignKeyValue1, ...]]
```
Comment (gerner-spryker) [inline src/Spryker/Zed/EventBehavior/Business/EventBehaviorFacade.php]: Please do not mix examples with descriptions. It's nice to have examples but they need to be in the long-description section. For return type and parameter type descriptions, you should use only describing explanations.

Could you please provide an input + output example? It's a bit complex what happens here. Probably a real example with `product` or something would be better than an abstract example.

File: src/Spryker/Zed/EventBehavior/Business/EventBehaviorFacadeInterface.php [hunk 2]
```diff
@@ -94,6 +94,20 @@ public function executeResolvedPluginsBySources(array $resources, array $ids = [
...
+     * - Returns an array of foreign keys grouped by column.
+     *
+     * @api
+     *
+     * @param \Generated\Shared\Transfer\EventEntityTransfer[] $eventTransfers
+     * @param string $foreignKeyColumnName
+     *
+     * @return array ['foreignKey' => ['relatedForeignKeyName1' => relatedForeignKeyValue1, ...]]
+     */
+    public function getGroupedEventTransferForeignKeysByColumn(array $eventTransfers, string $foreignKeyColumnName): array;
```
Comment (gerner-spryker) [inline src/Spryker/Zed/EventBehavior/Business/EventBehaviorFacadeInterface.php]: @ehsanmx Please verify if the facade naming fits

File: src/Spryker/Zed/EventBehavior/Business/EventBehaviorFacade.php (hunk 1 above)
Comment (Compixx) [reply src/Spryker/Zed/EventBehavior/Business/EventBehaviorFacade.php]: `getGroupedEventTransferForeignKeysByColumn` receives $eventTransfers and $foreignKeyColumnName, which value will be used for grouping. On the screenshot below, we receive eventTransfers after unassign product from stores. In a group method I group all foreign keys by fk_product_abstract. It is important to keep abstract product to store relation to unassign proper concrete products
https://cl.ly/3e0bede38593

File: src/Spryker/Zed/EventBehavior/Business/EventBehaviorFacadeInterface.php (hunk 2 above)
Comment (ehsanmx) [reply src/Spryker/Zed/EventBehavior/Business/EventBehaviorFacadeInterface.php]: I would go for `getEventTransfersGroupedByForeignKeys`

Comment (ehsanmx) [reply src/Spryker/Zed/EventBehavior/Business/EventBehaviorFacadeInterface.php]: Please also move this method close to other `getEventTransfers....`

Comment (gerner-spryker) [reply src/Spryker/Zed/EventBehavior/Business/EventBehaviorFacadeInterface.php]: we changed it to `getGroupedEventTransferForeignKeysByForeginKey`

**Pull request merged**